src/function/scalar/list/list_select.cpp

```diff
@@ -0,0 +1,103 @@
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/planner/expression_binder.hpp"
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/planner/expression/bound_parameter_expression.hpp"
+
+namespace duckdb {
+
+static void ListSelectFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	Vector &list = args.data[0];
+	Vector &selection_list = args.data[1];
+	idx_t count = args.size();
+
+	ListVector::Reserve(result, ListVector::GetListSize(selection_list));
+
+	bool all_constant = true;
+	for (idx_t i = 0; i < args.ColumnCount(); i++) {
```

Comment:
Nit:
```cpp
auto all_constant = (args.data[0].GetVectorType() == VectorType::CONSTANT_VECTOR) &&
   (args.data[1].GetVectorType() == VectorType::CONSTANT_VECTOR);
```

Reply:
Is `args.AllConstant()` a better solution?